datajunction-server/datajunction_server/construction/build_v2.py

```diff
@@ -981,7 +998,7 @@ async def build_ast(  # pylint: disable=too-many-arguments,too-many-locals
                     build_criteria=build_criteria,
                 ),
             )
-            if not physical_table:
+            if not physical_table or not use_materialized:
```

Comment:
Can this logic actually be moved slightly earlier, so that we don't retrieve the physical table at all if `use_materialized` is set to `False`?

Reply:
Done!